ios/platform-features/auto-layout/programmatical-constraints.workbook

```diff
@@ -0,0 +1,120 @@
+var orangeLabel = new UILabel () {
+        Text = "Orange",
+        TextColor = UIColor.White,
+        TextAlignment = UITextAlignment.Center,
+        BackgroundColor = UIColor.Orange
+};
+
+// This could be any parent view you add UI elements to.
+var rootView = RootViewController.View;
+rootView.AddSubview (orangeLabel);
+```
+
+#### Absolute coordinates
+
+Let’s position the `orangeLabel` at the top left corner manualy first so we can see it.
+
+```csharp
+orangeLabel.Frame = new CGRect (20, 20, 120, 50);
+```
+
+If we wanted to center this label in the middle of the screen we could, *technically*, do it using **absolute** x,y coordinates but this is **not** the most optimal, responsive and simple (will imply quite a bit of math) way to do it.
+
+#### Begin with Constaints
+
+`TranslatesAutoresizingMaskIntoConstraints` is **key** to using manual constraints.
+
+By default this is set to `true` which lets the system create a set of constraints for you.
+
+However it prevents you from adding **additional constraints**, see API doc:
+
+> Note that the autoresizing mask constraints fully specify the view’s size and position; therefore, you cannot add additional constraints to modify this size or position without introducing conflicts. If you want to use Auto Layout to dynamically calculate the size and position of your view, you must set this property to false, and then provide a non ambiguous, nonconflicting set of constraints for the view.
+
+```csharp
+orangeLabel.TranslatesAutoresizingMaskIntoConstraints = false;
+```
+
+Setting `TranslatesAutoresizingMaskIntoConstraints` to `false` should result in the label disappearing because we did not, *yet*, add our manual constraints.
+
+*Note: at this point the `Frame` we set before for `orangeLabel` is obsolete, it’s not used by the system anymore. You can try to comment the line where we set it, it won’t affect the constraints we’ll define next.*
+
+So let’s add some constraints.
+
+We can use **anchors** to align UI elements. You can use these constraints to programatically define your layout using Auto Layout.
+
+```csharp
+// This translates to: orangeLabel's CenterYAnchor (vertical alignment) = rootView's CenterYAnchor
+orangeLabel.CenterYAnchor.ConstraintEqualTo (rootView.CenterYAnchor).Active = true;
+```
+
+We can also use some **guides** to help us with the alignment.
+
+`UIView` provides a `LayoutMarginsGuide` property to represent the margins.
+
+```csharp
+var marginGuide = RootViewController.View.LayoutMarginsGuide;
+orangeLabel.LeadingAnchor.ConstraintEqualTo (marginGuide.LeadingAnchor).Active = true;
+```
+
+Finally we can restore the label’s **width** and **height**.
+
+```csharp
+orangeLabel.WidthAnchor.ConstraintEqualTo (120).Active = true;
+orangeLabel.HeightAnchor.ConstraintEqualTo (50).Active = true;
+```
+
+Feel free to play around with the constraints above to try to center the label vertically **and** horizontally.
+
+#### Purple label constrained to orange label
```

Comment:
This isn't a big issue, but H4 renders super similar to text in workbooks. Maybe worth bumping up all your headings by one?